charts/prefect-server/templates/deployment.yaml

```diff
@@ -76,11 +72,25 @@ spec:
           env:
             - name: HOME
               value: /home/prefect
+            - name: PREFECT_API_URL
+              value: {{ .Values.server.prefectApiUrl | quote }}
```

Comment:
we were **not** previously setting this, which was incorrect behavior. All other URLs are inferred from this, so it **needs** to be set.